src/__tests__/incidents/Incidents.test.tsx

```diff
@@ -35,71 +37,58 @@ describe('Incidents', () => {
       components: { sidebarCollapsed: false },
     } as any)
 
-    let wrapper: any
-    await act(async () => {
-      wrapper = await mount(
+    function Wrapper({ children }: WrapperProps) {
+      return (
         <Provider store={store}>
           <MemoryRouter initialEntries={[path]}>
-            <titleUtil.TitleProvider>
-              <Incidents />
-            </titleUtil.TitleProvider>
+            <titleUtil.TitleProvider>{children}</titleUtil.TitleProvider>
           </MemoryRouter>
-        </Provider>,
+        </Provider>
       )
-    })
-    wrapper.find(Incidents).props().updateTitle = jest.fn()
-    wrapper.update()
+    }
 
-    return { wrapper: wrapper as ReactWrapper }
+    return rtlRender(<Incidents />, { wrapper: Wrapper })
   }
 
-  describe('title', () => {
-    it('should have called the useUpdateTitle hook', async () => {
-      await setup([Permissions.ViewIncidents], '/incidents')
-      expect(titleUtil.useUpdateTitle).toHaveBeenCalledTimes(1)
-    })
-  })
-
   describe('routing', () => {
     describe('/incidents/new', () => {
-      it('should render the new incident screen when /incidents/new is accessed', async () => {
-        const { wrapper } = await setup([Permissions.ReportIncident], '/incidents/new')
-
-        expect(wrapper.find(ReportIncident)).toHaveLength(1)
+      it('The new incident screen when /incidents/new is accessed', () => {
+        render([Permissions.ReportIncident], '/incidents/new')
+        expect(screen.getByRole('form', { name: /report incident form/i })).toBeInTheDocument()
       })
 
-      it('should not navigate to /incidents/new if the user does not have ReportIncident permissions', async () => {
-        const { wrapper } = await setup([], '/incidents/new')
-
-        expect(wrapper.find(ReportIncident)).toHaveLength(0)
+      it('should not navigate to /incidents/new if the user does not have ReportIncident permissions', () => {
+        const { container } = render([], '/incidents/new')
+        expect(container).toMatchInlineSnapshot(`<div />`)
       })
     })
 
     describe('/incidents/visualize', () => {
-      it('should render the incident visualize screen when /incidents/visualize is accessed', async () => {
-        const { wrapper } = await setup([Permissions.ViewIncidentWidgets], '/incidents/visualize')
-
-        expect(wrapper.find(VisualizeIncidents)).toHaveLength(1)
+      it('The incident visualize screen when /incidents/visualize is accessed', async () => {
+        const { container } = render([Permissions.ViewIncidentWidgets], '/incidents/visualize')
+        await waitForElementToBeRemoved(() => container.querySelector('.css-1jydqjy'))
```

Comment:
This seems like a very specific selector 🤔; how did we decide on this? What are the possible things we need to look out of in future in case something changes in the incidents visualize page? I ask only because I could not yet see where this selector comes from when I inspect the visualize page

Reply:
@nobrayner

Reply:
I think we are removing those `querySelector()` are remnants of the Enzyme

Reply:
Looks like a Spinner selector. For certain tests - i.e. you are checking that an element _doesn't_ exit - you have to wait for the spinner to be gone before you can assert anything meaningful. Otherwise... Well, it doesn't exist if there is only a spinner!

I'll clean this one up now.

Reply:
Fixed (to be merged)